### ISSUE TITLE
Lit search package command

### ISSUE THREAD

**Issue by homer242** [luvit/lit#56]
It would be nice to have a lit search <package> command, wouldn't it ?

For now, it's pretty obscure for a new guy like me to have a look about packages available for luvit 2.0. So, I think a search command and a website like npm will be awesome to recruit new enthousiast for this tool :)

**Comment by creationix** [comment]
Yep, I thought there was already a task for this.  I guess it's on the wrong repo.

**Comment by creationix** [comment]
FWIW, you can browse the packages at lit.luvit.io if you have the jsonview extension in your browser.

**Comment by james2doyle** [comment]
Could there maybe be a `http://lit.luvit.io/database` that returns the JSON tree of all packages? Or maybe a `http://lit.luvit.io/export` that triggers the DB export to a JSON file?

**Comment by creationix** [comment]
The metadata itself isn't very easy to extract currently.  You need to walk the git graph and look for `package.lua` or `init.lua` for trees and look in the root blob for single-file packages.  Then you need a lua parser with a custom sandbox to extract the data.

I think I'll make a copy of the metadata on publish into the tag body as JSON.  Then it will be much easier to extract this data and perform queries.  Also I can document fields for things like homepage and docs so we can link to descriptions and READMEs for the packages.

**Comment by creationix** [comment]
Depends on #67

**Comment by creationix** [comment]
I just pushed an update to lit with a revamped search engine.

You can test it at https://luvit.io/lit.html

There are 4 kinds of filters (tag, author, name, and search), each can have
multiple terms that are ORed together.  The filters themselves are ANDed
together.

The `tag`, `author`, and `name` filters require a prefix like
`author:creationix` or `name:msgpack`.  If there is no prefix, it's assumed to
be a search term.

So `author:luvit` will show all packages by the luvit author.  But `author:luvit
stream` will show all packages by luvit *and* containing the search term
`stream`.  Prefixed terms are exact matches, use `*` to allow wildcards.  The
unprefixed search term is full-text search in name, keywords, description, and tags.

Also you can wrap terms in quotes to include spaces.  For example searching for
`"blue apple"` is one term `blue apple` while `blue apple` is two terms `blue`
and `apple`.

I don't like having both keywords and tags.  Should we encourage packages to
migrate towards "tags" or just use the "keywords" field internally for tag
searches?

There was one breaking change where the result "type" field used to be either
"package" or "author" and is now "directory" or "file" depending on if the
package is a single file or a while tree.

**Comment by homer242** [comment]
very nice ! well done !
